src/libraries/System.Collections.Immutable/src/System/Collections/Frozen/FrozenSet.cs

```diff
@@ -127,10 +127,10 @@ public abstract class FrozenSet<T> : ISet<T>,
 
         /// <summary>Gets a collection containing the values in the set.</summary>
         /// <remarks>The order of the values in the set is unspecified.</remarks>
-        public ImmutableArray<T> Items => ItemsCore;
+        public ImmutableArray<T> Items => ImmutableArrayFactory.Create(ItemsCore);
 
         /// <inheritdoc cref="Items" />
-        private protected abstract ImmutableArray<T> ItemsCore { get; }
+        private protected abstract T[] ItemsCore { get; }
```

Comment:
Why does ItemsCore on FrozenSet change to return T[] but KeysCore/ValuesCore on FrozenDictionary continue to return ImmutableArray?

Reply:
This is because only Int32FrozenSet.cs needs access to the array that was hidden inside of ImmutableArray<T> (the ImmutableArray<T>.array field). None of the other implementation types need access to this array directly.

I can make it all consistent if you like and have the ItemsCore return raw arrays everywhere.

Reply:
> I can make it all consistent if you like and have the ItemsCore return raw arrays everywhere.

Thanks, yeah, let's keep it consistent.

Reply:
Done.